src/main/java/games/chess/actions/MovePiece.java

```diff
@@ -0,0 +1,111 @@
+package games.chess.actions;
+
+import core.AbstractGameState;
+import core.actions.AbstractAction;
+import games.chess.ChessGameState;
+import games.chess.components.ChessPiece;
+
+import java.util.Objects;
+
+public class MovePiece extends AbstractAction {
+
+    private final int startX;
+    private final int startY;
+    private final int targetX;
+    private final int targetY;
+
+    public MovePiece(int sx, int sy, int tx, int ty) {
+        this.startX = sx;
+        this.startY = sy;
+        this.targetX = tx;
+        this.targetY = ty;
+    }
+
+    @Override
+    public boolean execute(AbstractGameState ags) {
+        ChessGameState gs = (ChessGameState) ags;
+        ChessPiece piece = gs.getPiece(startX, startY);
+
+        //Delete the piece in the start position
+        gs.deletePiece(piece); // Remove the piece from its original position
+
```

Comment:
What happens if there is no piece there? We should throw an error or do something?